database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala

```diff
@@ -180,11 +195,13 @@ trait WorkflowMetadataSummaryEntryComponent {
       excludeLabelOrValues,
       submissionTimestampOption,
       startTimestampOption,
-      endTimestampOption
+      endTimestampOption,
+      includeSubworkflows
     )
     val query = workflowMetadataSummaryEntries.filter(filter)
     (page, pageSize) match {
       case (Some(p), Some(ps)) => query.drop((p - 1) * ps).take(ps)
+      case (None, Some(ps)) => query.take(ps)
```

Comment:
Warning: Scanning through the code there are other cases that expect BOTH params. I would move this logic out of the database layer. At a higher level *outside* of the database there should be business logic that sets the default page size if `page` is selected, and selects the default page if only `pageSize` is selected.

Here is another case that uses both as `Some` (this code is broken anyway and should be deleted):

https://github.com/broadinstitute/cromwell/blob/c70505bc0b8e4cb35a88d1cb89a6917f57f835ca/engine/src/main/scala/cromwell/webservice/metadata/WorkflowQueryPagination.scala#L39

And here's one that triggers just on `page` instead of `pageSize`:

https://github.com/broadinstitute/cromwell/blob/c70505bc0b8e4cb35a88d1cb89a6917f57f835ca/services/src/main/scala/cromwell/services/metadata/impl/MetadataDatabaseAccess.scala#L214

Reply:
I've removed the bad headers and made `page` and `pageSize` public because it's useful, we rely on it in JMUI, and so we don't want it to accidentally get "nobody uses this"'d.

If it's wrong, that's cause for a ticket, not a "hide it under the rug" IMO and whatever query does end up looking like, it needs to include pagination.

Reply:
This behavior also matches what WES looks like it would expect

Reply:
ToL: There may be others places in the code with biz logic embedded in data access code. I'm not sure at the moment. I still believe this biz logic should be concentrated in _one_ place higher up, not sprinkled throughout DAOs.